### ISSUE TITLE
Jobs in Torque/CentOS (ParallelCluster) queuing but not running 

### ISSUE THREAD

**Issue by afernandezody** [aws/aws-parallelcluster#875]
**Environment:**
 - AWS ParallelCluster 
 - OS: Centos7
 - Scheduler: Torque
 - Master instance type: t2.micro
 - Compute instance type: c5.large

**Bug description and how to reproduce:**
Torque and CenOS are not working well together as jobs are queued but don't run. The error is similar to that reported in (centos7 and torque not working with 1.3.1 #135) but I'm using ParallelCluster.
My script looks like:
#!/bin/bash
#PBS -l nodes=2:ppn=2,walltime=30:00
#PBS -N ***
#PBS -j oe
mpirun -np 4 /home/centos/***
I've tried several other combinations but it doesn't make any difference.

**Additional context:**
Status:
[centos@ip-172-31-1-43 ~]$ pbsnodes
ip-172-31-1-43
     state = down
     power_state = Running
     np = 1000
     ntype = cluster
     note = MasterServer
     mom_service_port = 15002
     mom_manager_port = 15003

ip-172-31-11-93
     state = down,offline,MOM-list-not-sent
     power_state = Running
     np = 2
     ntype = cluster
     mom_service_port = 15002
     mom_manager_port = 15003

**Comment by lukeseawalker** [comment]
Hi @afernandezody,
we are aware of the issue but yet not provided any resolution.
I can suggest to switch OS as a workaround.

**Comment by eewolfe** [comment]
CentOS 7

I was wondering if you could point me in the right direction for this problem I am having with AWS Parallel Cluster. I have diagnosed and locally patched the problem when it was in cfncluster, and the problem exists in Parallel Cluster.

What happens is that when a node is scaled up an entry is made into a specific SQS queue, and sqswatcher on the master node picks this up and the addHost python function gets called to add the node to the PBS server by calling qmgr. Here is the place . . .

def addHost(hostname, cluster_user, slots, max_cluster_size):

    log.info('Adding %s with %s slots' % (hostname, slots))

    command = ("/opt/torque/bin/qmgr -c 'create node %s np=%s'" % (hostname, slots))
    run_command(command, log, raise_on_error=False)

There seems to be some disconnect between when the hostname of the new node gets resolved as the PBS server does not understand the short hostname that gets passed to it, and the command looks OK here . . . 

2019-04-19 00:17:23,938 INFO [sqswatcher:_parse_sqs_messages] Processing COMPUTE_READY event for instance i-0e949a77e669315b0
2019-04-19 00:17:23,938 INFO [torque:addHost] Adding ip-172-16-21-111 with 36 slots
Error marking node ip-172-16-21-111 - The updated okclients list hasn't been sent out to the cluster yet. The node(s) will automatically update to free once this is done and the mom has reported in.
2019-04-19 00:17:23,974 WARNING [utils:_run_command] Command '['/opt/torque/bin/pbsnodes', '-c', 'ip-172-16-21-111']' returned non-zero exit status 51
2019-04-19 00:17:23,975 INFO [torque:addHost] Connecting to host: ip-172-16-21-111 iter: 0

Checking the PBS server log you see . . . 

4/19/2019 00:18:07.465;01;PBS_Server.58391;Svr;PBS_Server;LOG_ERROR::get_node_from_str, Node ip-172-16-21-111 is reporting on node ip-172-16-21-111.ec2.internal, which pbs_server doesn't know about
04/19/2019 00:18:52.506;01;PBS_Server.58391;Svr;PBS_Server;LOG_ERROR::get_node_from_str, Node ip-172-16-21-111 is reporting on node ip-172-16-21-111.ec2.internal, which pbs_server doesn't know about
04/19/2019 00:18:54.233;04;PBS_Server.58383;Svr;check_nodes_work;node ip-172-16-21-111 not detected in 1555633134 seconds, marking node down

PBS sees the added node as down because the server did not recognize it as a valid hostname . . . 

root@ip-172-16-21-44 ~]# pbsnodes -a
ip-172-16-21-44
     state = down,offline
     power_state = Running
     np = 1000
     ntype = cluster
     note = MasterServer
     mom_service_port = 15002
     mom_manager_port = 15003

ip-172-16-21-111
     state = down
     power_state = Running
     np = 36
     ntype = cluster
     mom_service_port = 15002
     mom_manager_port = 15003

Here are the before and after results from sqswatcher when adding a compute node, by simply updating the hostname passed to addHost by adding domain name postfix . . . 

Before:
def addHost(hostname, cluster_user, slots, max_cluster_size):

    log.info('Adding %s with %s slots' % (hostname, slots))

After: 
def addHost(hostname, cluster_user, slots, max_cluster_size):

    hostname += '.ec2.internal'
    log.info('Adding %s with %s slots' % (hostname, slots))

Here is what the output looks like when a new compute host is added successfully . . . 

2019-04-18 23:28:44,504 INFO [sqswatcher:_parse_sqs_messages] Processing COMPUTE_READY event for instance i-0e7116252f7260926
2019-04-18 23:28:44,505 INFO [torque:addHost] Adding ip-172-16-21-31.ec2.internal with 36 slots
Error marking node ip-172-16-21-31.ec2.internal - The updated okclients list hasn't been sent out to the cluster yet. The node(s) will automatically update to free once this is done and the mom has reported in.
2019-04-18 23:28:44,533 WARNING [utils:_run_command] Command '['/opt/torque/bin/pbsnodes', '-c', 'ip-172-16-21-31.ec2.internal']' returned non-zero exit status 51
2019-04-18 23:28:44,534 INFO [torque:addHost] Connecting to host: ip-172-16-21-31.ec2.internal iter: 0
2019-04-18 23:28:44,544 INFO [transport:_log] Connected (version 2.0, client OpenSSH_7.4)

Now PBS recognizes the new compute node . . . 

pbsnodes -a
ip-172-16-21-13
     state = down,offline
     power_state = Running
     np = 1000
     ntype = cluster
     note = MasterServer
     mom_service_port = 15002
     mom_manager_port = 15003

ip-172-16-21-31.ec2.internal
     state = free
     power_state = Running
     np = 36
     ntype = cluster
     status = rectime=1555630583,macaddr=12:79:ab:c8:5c:52,cpuclock=Fixed,varattr=,jobs=,state=free,netload=59146657,gres=,loadave=0.00,ncpus=36,physmem=61661100kb,availmem=60817920kb,totmem=61661100kb,idletime=666,nusers=1,nsessions=1,sessions=19557,uname=Linux ip-172-16-21-31 3.10.0-957.10.1.el7.x86_64 #1 SMP Mon Mar 18 15:06:45 UTC 2019 x86_64,opsys=linux
     mom_service_port = 15002
     mom_manager_port = 15003

How can we get this investigated/fixed in AWS Parallel Cluster?

**Comment by sean-smith** [comment]
Hi @eewolfe,

Thanks for the detailed writeup! The solution you have will work for `us-east-1` but for other regions the FQDN is ` ip-12-34-56-78.[region].compute.internal` so that needs to be patched. Adding the suffix is a good short term fix but ultimately everything should work with the shortened domain name.

**Comment by demartinofra** [comment]
The fix for the issue related to centos 7 and torque scheduler has been merged into develop branch and will be released with the next version of AWS ParallelCluster. Resolving..
